python/pip_install/extract_wheels/lib/namespace_pkgs.py

```diff
@@ -23,19 +23,22 @@ def implicit_namespace_packages(
     namespace_pkg_dirs: Set[str] = set()
     standard_pkg_dirs: Set[str] = set()
     # Traverse bottom-up because a directory can be a namespace pkg because its child contains module files.
+    directory = pathlib.Path(directory)
+    ignored_dirnames = [pathlib.Path(p) for p in ignored_dirnames or ()]
     for dirpath, dirnames, filenames in os.walk(directory, topdown=False):
+        dirpath = pathlib.Path(dirpath)
```

Comment:
nit: easier to type-check with `mypy` if variable names don't  get reassigned with a type change. Could instead be `directory_path` (L26) and `for dirpath_str, dirnames, filenames`.